File: AspNetMvcUnleashed/0672329980_source/Chapter09/CS/MvcApplication1/MvcApplication1.Tests/Routes/RouteTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Web.Routing;
using MvcFakes;
using RouteDebugger.Routing;

namespace MvcApplication1.Tests.Routes
{
    [TestClass()]
    public class RouteTest
    {
        [TestMethod]
        public void DefaultRouteMatchesHome()
        {
            // Arrange
            var routes = new RouteCollection();
            MvcApplication.RegisterRoutes(routes);

            // Act
            var context = new FakeHttpContext("~/Home");
            var routeData = routes.GetRouteData(context);

            // Assert
            var matchedRoute = (NamedRoute)routeData.Route;
            Assert.AreEqual("Default", matchedRoute.Name);
        }


        

    [TestMethod]
    public void ProductInsertMatchesPost()
    {
        // Arrange
        var routes = new RouteCollection();
        MvcApplication.RegisterRoutes(routes);

        // Act
        var context = new FakeHttpContext("~/Product/Insert", "POST", false, null);
        var routeData = routes.GetRouteData(context);

        // Assert
        var matchedRoute = (NamedRoute)routeData.Route;
        Assert.AreEqual("ProductInsert", matchedRoute.Name);
    }




    [TestMethod] 
    public void ProductInsertDoesNotMatchGet()
    {
        // Arrange
        var routes = new RouteCollection();
        MvcApplication.RegisterRoutes(routes);

        // Act
        var context = new FakeHttpContext("~/Product/Insert", "GET", false, null);
        var routeData = routes.GetRouteData(context);

        // Assert
        if (routeData != null)
        {
            var matchedRoute = (NamedRoute)routeData.Route;
            Assert.AreNotEqual("ProductInsert", matchedRoute.Name);
        }
    }


    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter09/VB/MvcApplication1/MvcApplication1.Tests/Routes/RouteTest.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System.Web.Routing
Imports MvcFakes
Imports RouteDebugger.Routing

<TestClass()> _
Public Class RouteTest

    <TestMethod()> _
    Public Sub DefaultRouteMatchesHome()
        ' Arrange
        Dim routes = New RouteCollection()
        MvcApplication.RegisterRoutes(routes)

        ' Act
        Dim context = New FakeHttpContext("~/Home")
        Dim routeData = routes.GetRouteData(context)

        ' Assert
        Dim matchedRoute = CType(routeData.Route, NamedRoute)
        Assert.AreEqual("Default", matchedRoute.Name)
    End Sub


    <TestMethod()> _
    Public Sub ProductInsertMatchesPost()
        ' Arrange
        Dim routes = New RouteCollection()
        MvcApplication.RegisterRoutes(routes)

        ' Act
        Dim context = New FakeHttpContext("~/Product/Insert", "POST", False, Nothing)
        Dim routeData = routes.GetRouteData(context)

        ' Assert
        Dim matchedRoute = CType(routeData.Route, NamedRoute)
        Assert.AreEqual("ProductInsert", matchedRoute.Name)
    End Sub




    <TestMethod()> _
    Public Sub ProductInsertDoesNotMatchGet()
        ' Arrange
        Dim routes = New RouteCollection()
        MvcApplication.RegisterRoutes(routes)

        ' Act
        Dim context = New FakeHttpContext("~/Product/Insert", "GET", False, Nothing)
        Dim routeData = routes.GetRouteData(context)

        ' Assert
        If routeData IsNot Nothing Then
            Dim matchedRoute = CType(routeData.Route, NamedRoute)
            Assert.AreNotEqual("ProductInsert", matchedRoute.Name)
        End If
    End Sub



End Class
